headers/augment_headers.py

```diff
@@ -155,12 +158,14 @@ def _input_header_lines(lines) -> Iterator[Tuple[str, bool, bool]]:
             if not in_comment:
                 in_docstring = False
 
-            yield line, in_comment, in_docstring
+            is_deprecated_macro = not in_comment and line.startswith("DEPRECATED(")
+
+            yield line, in_comment, in_docstring, is_deprecated_macro
```

Comment:
I know this is getting a bit unwieldy, but can we also add an `in_directive` check for preprocessor directives? Otherwise we'll try to match the `#define DEPRECATED(` line against the function regex and it'll probably yield the false symbol name "DEPRECATED". So something like

```py
in_directive = not in_comment and line.lstrip().startswith("#")
is_deprecated_macro = not in_comment and line.lstrip().startswith("DEPRECATED(")

yield line, in_comment, in_docstring, in_directive, is_deprecated_macro
```